### PR TITLE
Skip local GraalVM artifacts when not building from source

### LINKED ISSUE #570: CI: Quarkus build fails when using builder images.
## Description
https://github.com/graalvm/mandrel/issues/526 adds support for building with locally installed graalvm maven artefacts. This seems to break builds using builder images, which don't build mandrel/graal from source. The CI check needs to skip downloading those artefacts when an builder image CI build is being triggered.

See this example failure:
```
Starting download for org-graalvm-artefacts--2.13.8.Final-quarkusio-quarkus-graalvm-mandrel-graal-master-master-graalvm-mandrel-packaging-mandrel-source-17-ga-quay.io-quarkus-ubi-quarkus-mandrel-builder-image-22.3-java17-null-graalvm-mandrel-null-null
Error: Unable to find any artifacts for the associated workflow
```
https://github.com/graalvm/mandrel/actions/runs/6168721678/job/16741701212#step:7:24

### PR DESCRIPTION
**Pull request by zakkak**
Closes: https://github.com/graalvm/mandrel/issues/570

### CODE REVIEW

File: .github/workflows/base.yml
Comment (jerboaa) [inline .github/workflows/base.yml]: `distribution == graalvm` builds those too and we should download them in that case. So it should be "mandrel or graalvm" whatever the syntax for that is.

File: .github/workflows/base.yml
Comment (jerboaa) [inline .github/workflows/base.yml]: Same, re: `graalvm`

File: .github/workflows/base.yml
Comment (jerboaa) [reply .github/workflows/base.yml]: Perhaps we can just drop the condition on `mandrel`. `builder image == null && build from source == true` should be sufficient?

File: .github/workflows/base.yml
Comment (jerboaa) [inline .github/workflows/base.yml]: Need to account for graalvm too.

**Review by jerboaa — COMMENTED** [review]
Do we have a CI run testing this? At least the local build one for `mandrel` & `graalvm`.

File: .github/workflows/base.yml
Comment (zakkak) [reply .github/workflows/base.yml]: Yes

**Comment by zakkak** [comment]
> Do we have a CI run testing this? At least the local build one for `mandrel` & `graalvm`.

mandrel from source:  https://github.com/zakkak/mandrel/actions/runs/6185883989
graalvm from source: https://github.com/zakkak/mandrel/actions/runs/6185892130
quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-17: https://github.com/zakkak/mandrel/actions/runs/6185839831

File: .github/workflows/base-wrapper.yml
Comment (jerboaa) [inline .github/workflows/base-wrapper.yml]: I take it this will be reverted once CI completes?

**Review by jerboaa — APPROVED** [review]
LGTM. Please wait for CI before you merge. Thanks!

File: .github/workflows/base-wrapper.yml
Comment (zakkak) [reply .github/workflows/base-wrapper.yml]: Yes, I better not forget it :)

**Comment by zakkak** [comment]
CI runs are good, TEMP commit dropped. I am merging.

**Pull request merged**
